tests/src/Functional/GroupUpdateTest.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\og\Functional\GroupUpdateTest.
+ */
+
+namespace Drupal\Tests\og\Functional;
+
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\node\Entity\Node;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Entity\OgRole;
+use Drupal\og\Og;
+use Drupal\og\OgAccess;
+use Drupal\og\OgMembershipInterface;
+use Drupal\simpletest\ContentTypeCreationTrait;
+use Drupal\simpletest\NodeCreationTrait;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests the special permission 'update group'.
+ *
+ * @group og
+ */
+class GroupUpdateTest extends BrowserTestBase {
+
+  use ContentTypeCreationTrait;
+  use NodeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['node', 'entity_test', 'og'];
+
+  /**
+   * Test content group.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface
+   */
+  protected $content_group;
+
+  /**
+   * Test entity group.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface
+   */
+  protected $entity_group;
+
+  /**
+   * Test group owner user.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $group_owner;
+
+  /**
+   * Test group editor user.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $group_editor;
+
+  /**
+   * Test normal user with no connection to the organic group.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $normal_user;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create dummy users.
+    $this->group_owner = $this->drupalCreateUser();
+    $this->group_editor = $this->drupalCreateUser();
+    $this->normal_user = $this->drupalCreateUser();
+
+    $this->setUpContentEntity();
+    $this->setUpEntityTestEntity();
+  }
+
+  /**
+   * Setup dummy content group entity and appropriate og permissions.
+   */
+  public function setUpContentEntity() {
+    // Create a node bundle called 'content_group' and make it an Og group.
+    $this->createContentType(['type' => 'content_group']);
+    Og::groupManager()->addGroup('node', 'content_group');
+
+    // Create a role with only the 'update group' permission.
+    $content_editor_role = OgRole::create();
+    $content_editor_role
+      ->setName('content_editor')
+      ->setLabel('Content group editor')
+      ->setGroupType('node')
+      ->setGroupBundle('content_group')
+      ->grantPermission(OgAccess::UPDATE_GROUP_PERMISSION)
+      ->save();
+
+    // Create a group content owned by the group owner.
+    $values = [
+      'title' => 'My awesome content group',
```

Comment:
ow.. and I tried so much to come up with such a great title :D changed (after the push)